vpr/src/place/initial_placement.cpp

```diff
@@ -51,38 +51,214 @@ static vtr::vector<ClusterBlockId, t_block_score> assign_block_scores();
 //Sort the blocks according to how difficult they are to place, prior to initial placement
 static std::vector<ClusterBlockId> sort_blocks(const vtr::vector<ClusterBlockId, t_block_score>& block_scores);
 
-//Sort the macros according to how difficult they are to place, prior to initial placement
-static std::vector<t_pl_macro> sort_macros(const vtr::vector<ClusterBlockId, t_block_score>& block_scores);
+//Get a legal placement position for the block
```

Comment:
improve this comment (look for one random legal position for a block that respects its region constraints)

Add comments for exhaustive placement routines as well

Reply:
describe what it returns (true/false based on whether it found a location) and the location by reference